src/network/dhcpwait.c

```diff
@@ -37,6 +37,11 @@ void await_dhcp(void)
 	k_mutex_unlock(&dhcp_lock);
 }
 
+bool is_dhcp_up()
+{
+	return dhcp_running;
```

Comment:
I didn't really think about this before, and probably doesn't matter on the coretex CPUs we're on, but technically, this should be an atomic variable, or there could be risks of incoherence between the different threads reading and writing it.

Reply:
There is support for atomic variables in Zephyr, at least on Arm from memory 👍

Reply:
@d3zd3z What do you think is the best synchronization mechanism in this use case in Zephyr to indicate a state change across threads? Zephyr has several options for this with various pros and cons. I did look at the docs, but curious if you have a strong opinion here?